acme/acme/client.py

```diff
@@ -626,3 +627,43 @@ def post(self, url, obj, content_type=JSON_CONTENT_TYPE, **kwargs):
         response = self._send_request('POST', url, data=data, **kwargs)
         self._add_nonce(response)
         return self._check_response(response, content_type=content_type)
+
+    def set_user_agent(self, keyval_map):
```

Comment:
please move it to `letsencrypt` module, together with `user_agent_object`. It's not suitable for general purpose ACME library.